Examples/GenerativeAISample/GenerativeAIMultimodalSample/ViewModels/PhotoReasoningViewModel.swift

```diff
@@ -61,7 +64,17 @@ class PhotoReasoningViewModel: ObservableObject {
       var images = [PartsRepresentable]()
       for item in selectedItems {
         if let data = try? await item.loadTransferable(type: Data.self) {
-          images.append(ModelContent.Part.png(data))
+          guard let image = UIImage(data: data) else {
+            logger.error("Failed to parse data as an image, skipping.")
+            continue
+          }
+          guard let resizedImage = image
+            .preparingThumbnail(of: PhotoReasoningViewModel.maxImageSize) else {
```

Comment:
Does this method preserve the aspect ratio of the original image and size it to fit within `maxImageSize`? The [documentation](https://developer.apple.com/documentation/uikit/uiimage/3750835-preparingthumbnail) doesn't say anything.

Reply:
Good catch! `preparingThumbnail(of:)` doesn't maintain the aspect ratio and I added `fitting(maxSize: CGSize)` below as an extension on `CGSize` to work around this... but then forgot to use it.

The latest commit maintains the aspect ratio and avoids calling `preparingThumbnail(of:)` if the image is already smaller than `PhotoReasoningViewModel.maxImageSize`.